x-pack/test_utils/jest/contract_tests/servers.ts

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import url from 'url';
+import { resolve } from 'path';
+import * as kbnTestServer from '../../../../src/test_utils/kbn_server';
+import { TestKbnServerConfig } from '../../kbn_server_config';
+
+const PLUGIN_PATHS = resolve(__dirname, '../../../');
+
+interface ESServerConfig {
+  hosts: string[];
+  username: string;
+  password: string;
+}
+
+type ESServer = {
+  stop: () => void;
+} & ESServerConfig;
+
+let ESServer: ESServer;
+
+function _parseESConnectionString(connectionString: string) {
+  const { username, password } = (url.parse(connectionString) as unknown) as url.URL;
+
+  return {
+    hosts: [connectionString],
+    username,
+    password,
+  };
+}
+
+/**
+ * Create a new shared ES server,
+ * this function should not be used outside of jest globalSetup
+ */
+export async function _createSharedServer() {
+  if (process.env.ES_SERVER_URL) {
+    process.env.__JEST__ESServer = JSON.stringify(
```

Comment:
Do we need to use an env var here if its only being used within this file?

Reply:
Yes because we have a different context between the test execution and global setup, also jest automatically call a `resetModules` method so we can use any global variables here

Reply:
Can we please clear that up by accessing this variable with something like `setJestSafeGlobal()`/`getJestSafeGlobal()` that has a comment explaining the situation and why this is necessary?